src/Configurator/PhpCsConfigurator.php

```diff
@@ -0,0 +1,199 @@
+                        }
+                    }
+
+                    return $value;
+                }
+            )->markOptional() // Optional, as this should be configured using a defaults value.
+        );
+
+        $questions->communicate(
+            'php_cs_linting',
+            Question::confirm('Enable PHP-CS linting?', false)
+        );
+
+        if ($questions->communicate(
+            'styleci_enabled',
+            Question::confirm('Enable StyleCI (local configuration)?', true)
+        )
+        ) {
+            $questions->set('php_cs_version_1_bc', false);
+            $questions->communicate('php_cs_styleci_bridge', Question::confirm('Enable StyleCI bridge', true));
+        } else {
+            $questions->set('php_cs_styleci_bridge', false);
+            $questions->communicate('php_cs_version_1_bc', Question::confirm('Enable PHP-CS v1 compatibility?', true));
+        }
+
+        // Finder questions.
+        $questions->communicate('php_cs_finder_path', Question::ask('PHP-CS Finder {path}', 'src')->markOptional());
+        $questions->communicate('php_cs_finder_not_path', Question::ask('PHP-CS Finder {not path}')->markOptional());
+        $questions->communicate('php_cs_finder_exclude', Question::ask('PHP-CS Finder {exclude dirs}')->markOptional());
+
+        $questions->communicate('php_cs_finder_name', Question::ask('PHP-CS Finder {name}')->markOptional());
+        $questions->communicate('php_cs_finder_not_name', Question::ask('PHP-CS Finder {not name}')->markOptional());
+
+        $questions->communicate('php_cs_finder_contains', Question::ask('PHP-CS Finder {contains}')->markOptional());
+        $questions->communicate('php_cs_finder_not_contains', Question::ask('PHP-CS Finder {not contains}')->markOptional());
+
+        $questions->communicate('php_cs_finder_depth', Question::ask('PHP-CS Finder {depth}')->markOptional());
+    }
+
+    public function finalizeConfiguration(array &$configuration)
+    {
+        if ($configuration['php_cs_styleci_bridge']) {
+            $configuration['composer']['require-dev'][] = 'sllh/php-cs-fixer-styleci-bridge';
+        }
+
+        // First resolve the aliases to v2 (as they are used), then remove overlapping fixers
+        // Then when v1 support is enabled, resolve back to v1.
+
+        $fixers = $this->resolveAliases($this->splitValues($configuration['php_cs_enabled_fixers']));
+        $preset = strtolower($configuration['php_cs_preset']).'_fixers';
+
+        if (in_array($configuration['php_cs_preset'], ['laravel', 'recommended'], true)) {
+            $configuration['php_cs_enabled_fixers'] = array_unique(
+                array_merge($configuration['php_cs_enabled_fixers'], Fixers::${$preset})
+            );
+
+            $configuration['php_cs_level'] = 'none';
+        } else {
+            $configuration['php_cs_level'] = $configuration['php_cs_preset'];
+        }
+
+        if ('none' !== $configuration['php_cs_preset']) {
```

Comment:
Will fail when 'laravel' or 'recommended' is used, needs to check against `$configuration['php_cs_level']` instead.
And this is fragile for when using the styleci bridge (as the preset is not a problem then).

Consider refactoring this method.